### DISCUSSION TITLE
how to disable touchpad while "hyprctrl devices" show Touch nothing

### DISCUSSION THREAD

**Discussion by caigxg** [hyprwm/Hyprland#1561] [Q&A] [answered]
when I use command "hyprctrl devices" to see my touchpad ,it shows nothing ,
then how can I disable touchpad

**Comment by vaxerski** [comment, marked as answer]
touchpad is a mouse. Look into your mice

**Reply by caigxg** [reply]
thanks,I notice this two days ago.and it is ok now.
another , I config in my hyprland.conf :
bind = SUPER, SPACE, exec, rofi -show drun 
when I use Super+SPACE ,select intelli idea,it shows me need to active idea
but when I open kitty and exec rofi -show drun ,select intelli idea ,it is ok
I wonder what should I do to resolve this?

**Reply by vaxerski** [reply]
no clue, but it might be your bashrc not being sourced if you use a login manager. try launching hyprland from the tty

**Reply by caigxg** [reply]
I stop sddm,and it's ok .
I wonder to know what the different :
Method1.use sddm , config in hyprland.conf: exec-once = . ~/.bashrc
Method2.disable sddm,start hyprland in tty

~/.bashrc just export some variables,and these variables have values when use Method1

**Reply by vaxerski** [reply]
exec-once spawns a separate process. The variables do not propagate to Hyprland.

add `source /home/youruser/.bashrc` to your wrapper

**Reply by caigxg** [reply]
I also export variables in my wrapper in Method1 ,and I can see the variable's value.
When use short cut configed in hyprland to open rofi, then select Intelli idea,idea show me to active.
When use kitty to exec rofi,then select Intelli idea,it is ok
When use /opt/intellij-idea-ultimate-edition/bin/idea.sh to start idea,it is ok
so I don't understand

**Reply by vaxerski** [reply]
`source /home/you/.bashrc && rofi -show run`?
